### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by tomily1**
### What does this PR do?

This pull request was created to check houndCI

### Description of Task to be completed?

refactor of code and ensure no houndCI flag is raised

### How should this be manually tested?
`npm test`

### Any background context you want to provide?

No background context

### Screenshots (if appropriate)

screen shot 2017-01-06 at 9 01 48 am

### Questions:

Not Applicable.

### CODE REVIEW

File: src/string-class.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+' use strict';
+
+const StringClass = {
+  /**
+   * Tests if a string contains vowels
+   * @return {Boolean} True or false depending on the match
+   */
+  hasVowels() {
+    return /[aeiouAEIOU]/g.test(this);
```
Comment (femioladeji) [inline src/string-class.js]: Why not use the case-insensitive flag `i` so that you won't have to add AEIOU

File: src/string-class.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    return this.replace(/[A-Z]/g, (character) => {
+      return String.fromCharCode(character.charCodeAt() + 32);
+    });
+  },
+  /**
+   * converts strings to title case
+   * @return {String} string that has been converted to title case
+   */
+  ucFirst() {
+    return `${this.slice(0, 1).toUpper()}${this.slice(1).toLower()}`;
```
Comment (femioladeji) [inline src/string-class.js]: Does the instruction say all other characters should be converted to lower case?

File: src/string-class.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+   */
+  wordCount() {
+    return this.word().length;
+  },
+  /**
+   * converts currency representation to string
+   * @return {String} currency representation of the the number in String
+   */
+  toCurrency() {
+    let roundCurrency = String(Math.round(((Number(this)) + 0.00001) * 100) / 100);
```
Comment (femioladeji) [inline src/string-class.js]: This looks a bit complicated, you should probably check out `.toFixed()`. [Check here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed)

File: src/string-class.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      return /[A-Z]/.test(character) ? character.toLower() : character.toUpper();
+    });
+  },
+  /**
+   * converts each character of a String to alternating case
+   * @return {String} alternate case of each character of the String
+   */
+  alternatingCase() {
+    return this.replace(/\w/g, (character, index) => {
+      return (index + 1) % 2 === 0 ? character.toUpper() : character.toLower();
```
Comment (femioladeji) [inline src/string-class.js]: Why not just use `index % 2 !== 0` or  `index % 2 === 1` to prevent the extra 3 token of `+ 1`. It doesn't affect performance though

File: test/string-class-test.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    it('should return an unchanged string if first letter is uppercase', () => {
+      expect('Andela'.ucFirst()).to.equal('Andela');
+      expect('Tomilayo'.ucFirst()).to.equal('Tomilayo');
+    });
+  });
+
+  describe('isQuestion', () => {
+    it('should return "true" if the entire string is a question', () => {
+      expect('How are you?'.isQuestion()).to.be.true;
+    });
```
Comment (abdulsemiu-atanda) [inline test/string-class-test.js:63]: this fails when 'How are you?' is written as 'How are you??'. You may want to handle this edge case

File: src/string-class.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  },
+  /**
+   * checks for double characters in a string
+   * @return {Boolean} True of false depending on thne match
+   */
+  doubleCheck() {
+    return /(.)\1/.test(this);
+  }
+};
+Object.assign(String.prototype, StringClass);
```
Comment (abdulsemiu-atanda) [inline src/string-class.js]: looks good to me

File: public/index.html [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+    </div>
+
+    <footer>
+        <p class="footer">Copyright 2016 - Israel Tomilayo - Andela</p>
+    </footer>
+</body>
+<script>
+// Get the modal
+var modal = document.getElementById('myModal');
```
Comment (olawalejarvis) [inline public/index.html]: ### Nit-Pick
 - Use let statement 
 - I think it is best to separate your script from your view

File: public/src/string-class.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
+' use strict';
+
+const StringClass = {
+  /**
```
Comment (olawalejarvis) [inline public/src/string-class.js:4]: ### Nit-Pick
Kes want us to follow the below comment's format 

/**
 * Name of function
 *
 * Short description
 *
 * @return{...}
 */

File: public/src/string-class.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+   */
+  wordCount() {
+    return this.word().length;
+  },
+  /**
+   * converts currency representation to string
+   * @return {String} currency representation of the the number in String
+   */
+  toCurrency() {
+    let roundCurrency = Number(this).toFixed(2);
```
Comment (olawalejarvis) [inline public/src/string-class.js]: ### Nit-Pick
You may want it to return an error message instead of the default "NaN"

File: public/src/string-class.js [hunk 10]
```diff
@@ -0,0 +1,137 @@
+' use strict';
+
+const StringClass = {
+  /**
+   * Tests if a string contains vowels
+   * @return {Boolean} True or false depending on the match
+   */
+  hasVowels() {
+    return /[aeiou]/i.test(this);
+  },
```
Comment (ghost) [inline public/src/string-class.js:11]: ###Nitpick
Leave a space after each method for readability

File: test/string-class-test.js [hunk 11]
```diff
@@ -0,0 +1,175 @@
+var expect = require('chai').expect;
+require('../public/src/string-class');
+
+describe('String Class', () => {
+
+  describe('hasVowels', () => {
+    it('should return true for strings that has vowels in them', () => {
+      expect('word'.hasVowels()).to.be.true;
+      expect('andela'.hasVowels()).to.be.true;
```
Comment (olawalejarvis) [inline test/string-class-test.js]: ### Nit-Pick
Since you are testing the same thing for both line 8 and 9, you may want to remove one out of the two

Also in line 15 and 16, I think there is no need to repeat the same test twice, maybe you can remove one of it also

Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case

File: test/string-class-test.js [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+    it('should return false for strings with consonants only', () => {
+      expect('cyst'.hasVowels()).to.be.false;
+      expect('Hymn'.hasVowels()).to.be.false;
+    });
+  });
+
+  describe('toUpper', () => {
+    it('should convert strings in lowercase to uppercase', () => {
+      expect('cyst'.toUpper()).to.equal('CYST');
+      expect('andela'.toUpper()).to.equal('ANDELA');
```
Comment (olawalejarvis) [inline test/string-class-test.js]: ### Nit-Pick
Repeating the same test twice, you can remove one of the two (line 22 and line 23)
Also in line 26 and 27

Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case

File: test/string-class-test.js [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+    });
+    it('should convert strings with title case to uppercase', () => {
+      expect('Cyst'.toUpper()).to.equal('CYST');
+      expect('Andela'.toUpper()).to.equal('ANDELA');
```
Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case

File: test/string-class-test.js [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+    });
+    it('uppercase strings should remain in uppercase', () => {
+      expect('ANDELA'.toUpper()).to.equal('ANDELA');
+    });
+  });
+
+  describe('toLower', () => {
+    it('should convert strings in uppercase to lowercase', () => {
+      expect('ANDELA'.toLower()).to.equal('andela');
+      expect('TOMILAYO'.toLower()).to.equal('tomilayo');
```
Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case.

File: test/string-class-test.js [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+    });
+    it('should convert strings with title case to lowercase', () => {
+      expect('Andela'.toLower()).to.equal('andela');
+      expect('Tomilayo'.toLower()).to.equal('tomilayo');
```
Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case.

File: test/string-class-test.js [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+  describe('toCurrency', () => {
+    it('should return a currency representation of a number', () => {
+      expect('10000.00'.toCurrency()).to.equal('10,000.00');
+    });
+    it('should return a currency representation with decimals', () => {
+      expect('5000'.toCurrency()).to.equal('5,000.00');
+    });
+    it('should return a currency representation with two decimals', () => {
+      expect('123456.78945'.toCurrency()).to.equal('123,456.79');
+    });
```
Comment (olawalejarvis) [inline test/string-class-test.js:100]: ### Nit-Pick
You may want to test for invalid input

File: test/string-class-test.js [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+      expect('Tomilayo'.toLower()).to.equal('tomilayo');
+    });
+    it('should convert strings that are mixed with both lowercase and uppercase to lowercase', () => {
+      expect('cYsT'.toLower()).to.equal('cyst');
+      expect('AnDelA'.toLower()).to.equal('andela');
+      expect('ToMilaYO'.toLower()).to.equal('tomilayo');
+    });
+    it('lowercase strings should remain in lowercase', () => {
+      expect('andela'.toLower()).to.equal('andela');
+      expect('tomilayo'.toLower()).to.equal('tomilayo');
```
Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case.

File: test/string-class-test.js [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+  });
+
+  describe('ucFirst', () => {
+    it('should return the string with the first letter in upper case', () => {
+      expect('AnDelA'.ucFirst()).to.equal('AnDelA');
+      expect('tomilayo'.ucFirst()).to.equal('Tomilayo');
+    });
+    it('should return an unchanged string if first letter is uppercase', () => {
+      expect('Andela'.ucFirst()).to.equal('Andela');
+      expect('Tomilayo'.ucFirst()).to.equal('Tomilayo');
```
Comment (ghost) [inline test/string-class-test.js]: ###Critical
Remove redundant test case.

File: test/string-class-test.js [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+  describe('isDigit', () => {
+    it('should return "true" if string is a digit', () => {
+      expect('7'.isDigit()).to.be.true;
+    });
+    it('should return "false" if string is not a single digit', () => {
+      expect('77'.isDigit()).to.be.false;
+    });
+    it('should be able to handle negative and positive operators', () => {
+      expect('-7'.isDigit()).to.be.true;
+      expect('+7'.isDigit()).to.be.true;
```
Comment (ghost) [inline test/string-class-test.js]: ### Critical
By definition, a digit is any single number between 0 and 9. Adding operators does not make them digits anymore.

File: test/string-class-test.js [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+    });
+    it('should return an unchanged string if first letter is uppercase', () => {
+      expect('Andela'.ucFirst()).to.equal('Andela');
+      expect('Tomilayo'.ucFirst()).to.equal('Tomilayo');
+    });
+  });
+
+  describe('isQuestion', () => {
+    it('should return "true" if the entire string is a question', () => {
+      expect('How are you?'.isQuestion()).to.be.true;
```
Comment (ghost) [inline test/string-class-test.js:62]: Write a test to check if the statement ends with any other character other than '?' e.g "I love biscuits."
